processor/src/stack/overflow.rs

```diff
@@ -0,0 +1,92 @@
+use super::Felt;
+use winter_utils::collections::BTreeMap;
+
+// OVERFLOW
+// ================================================================================================
+
+/// Overflow table stores a list of overflow values from the stack.
+/// We create a separate structure to be able to store historical changes
+/// in the overflow table, so we are able to reconstruct what's in the overflow table
+/// at any given point in time.
+#[derive(Clone)]
+pub struct OverflowTable {
+    active: Vec<Felt>,
+    trace: BTreeMap<usize, Vec<Felt>>,
+    trace_enabled: bool,
+}
+
+impl OverflowTable {
+    pub fn new(trace_enabled: bool) -> Self {
+        Self {
+            active: Vec::new(),
+            trace: BTreeMap::new(),
+            trace_enabled,
+        }
+    }
+
+    pub fn push(&mut self, step: usize, value: Felt) {
+        self.active.push(value);
+        if self.trace_enabled {
+            self.trace.insert(step, self.active.clone());
+        }
+    }
+
+    pub fn pop(&mut self, step: usize) -> Option<Felt> {
+        let popped = self.active.pop();
+        if popped.is_some() && self.trace_enabled {
+            self.trace.insert(step, self.active.clone());
+        }
+
+        popped
+    }
+
+    /// Add values up to a certain step from the overflow table.
+    pub fn add_values_at(&self, step: usize, values: &mut Vec<Felt>) {
+        if let Some(x) = self.trace.range(0..=step).last() {
+            values.extend_from_slice(x.1);
+        }
+    }
+
+    /// Add values from the overflow table into a target vector.
+    pub fn add_values(&self, vec: &mut Vec<Felt>) {
+        vec.extend_from_slice(self.active.as_slice());
+    }
```

Comment:
nit: maybe rename to `append_into()`.

We can also take an extra parameter here to indicate the number of items to be appended. This will make the logic simpler in `Stack::get_values()`

Reply:
Yes this was the original change, I think you suggested we can remove this, I''ll add this back

Reply:
fixed